src/mjolnir/osmdata.cc

```diff
@@ -308,6 +320,31 @@ bool write_linguistic(const std::string& filename, const LinguisticMultiMap& lin
   return true;
 }
 
+bool write_conditional_speed_limits(const std::string& filename,
+                                    const ConditionalSpeedLimitsMultiMap& speed_map) {
+  // Open file and truncate
+  std::ofstream file(filename.c_str(), std::ios::out | std::ios::binary | std::ios::trunc);
+  if (!file.is_open()) {
+    LOG_ERROR("write_linguistic failed to open output file: " + filename);
+    return false;
+  }
+
+  // Convert the multi map into a flat representation
+  std::vector<TempConditionalSpeedLimit> flat;
+  flat.reserve(speed_map.size());
+  for (auto it = speed_map.cbegin(); it != speed_map.cend(); ++it) {
+    flat.emplace_back(it->first, it->second);
+  }
+
+  // Write the count and then the via ids
+  uint32_t sz = flat.size();
+  file.write(reinterpret_cast<const char*>(&sz), sizeof(uint32_t));
+  file.write(reinterpret_cast<const char*>(flat.data()),
+             flat.size() * sizeof(TempConditionalSpeedLimit));
+  file.close();
```

Comment:
sure i see others have followed this pattern but we can do a little better. you can avoid having to copy all the data before writing it and can even avoid having the temporary structure both by simply writing it directly to the file:

```suggestion
  // Convert the multi map into a flat representation
  uint32_t sz = speed_map.size();
  file.write(reinterpret_cast<const char*>(&sz), sizeof(uint32_t));
  for(const auto& sp : speed_map) {
    file.write(reinterpret_cast<const char*>(&sp.first), sizeof(sp.first));
    file.write(reinterpret_cast<const char*>(&sp.second), sizeof(sp.second));
  }
  file.close();
```

Reply:
Added in b08327ce2b9c8cbbbf0ac0bd8826a39f096baa17